pkg/controller/endpointslice/reconciler_test.go

```diff
@@ -71,6 +71,12 @@ func TestReconcile1Pod(t *testing.T) {
 	namespace := "test"
 	ipv6Family := corev1.IPv6Protocol
 	svcv4, _ := newServiceAndEndpointMeta("foo", namespace)
+	svcv4ClusterIP, _ := newServiceAndEndpointMeta("foo", namespace)
+	svcv4ClusterIP.Spec.ClusterIP = "1.1.1.1"
+	svcv4Labels, _ := newServiceAndEndpointMeta("foo", namespace)
+	svcv4Labels.Labels = map[string]string{"foo": "bar"}
+	svcv4BadLabels, _ := newServiceAndEndpointMeta("foo", namespace)
+	svcv4BadLabels.Labels = map[string]string{discovery.LabelServiceName: "bad", discovery.LabelManagedBy: "actor"}
```

Comment:
Mind adding something like `corev1.IsHeadlessService: "invalid"` here as well?

Reply:
yeah, no prob, but is that value of that label meaningful?

Reply:
Nope, can be anything, just want to ensure a malicious value on Service doesn't impact EndpointSlices.

Reply:
done